test/jest/default-request-executor.test.js

```diff
@@ -0,0 +1,406 @@
+
+    it('should build a new request and send it to this.fetch', async () => {
+      const requestExecutor = new DefaultRequestExecutor();
+      const mockRequest = { method: 'GET' };
+      requestExecutor.fetch = jest.fn();
+      const now = new Date();
+      const retryAfter = new Date(now.getTime() + 1000); // one second in the future
+      const mockResponse = buildMockResponse({
+        headers: {
+          'x-rate-limit-reset' : String(dateToEpochSeconds(retryAfter)),
+          date: now.toUTCString()
+        }
+      });
+      /*
+        retryRequest will call buildRetryRequest to build the object that it
+        sends to this.fetch(), so we use buildRetryRequest() to get an object
+        that is similar to what we expect to be sent to fetch().  But we can't
+        directly compare the startTime property, because it may differ between
+        the two objects.
+      */
+      const expectedNewRequest = requestExecutor.buildRetryRequest(mockRequest, mockResponse);
+      await requestExecutor.retryRequest(mockRequest, mockResponse);
+      const actualNewRequest = requestExecutor.fetch.mock.calls[0][0];
+      expect(Math.abs(expectedNewRequest.startTime - actualNewRequest.startTime)).toBeLessThan(1000);
+      delete expectedNewRequest.startTime;
+      delete actualNewRequest.startTime;
+      expect(expectedNewRequest).toEqual(actualNewRequest);
+    });
+
+    it('should reject if fetch rejects', async () => {
+      const requestExecutor = new DefaultRequestExecutor();
+      const mockRequest = { method: 'GET' };
+      const now = new Date();
+      const retryAfter = new Date(now.getTime() + 1000); // one second in the future
+      const mockResponse = buildMockResponse({
+        headers: {
+          'x-rate-limit-reset' : String(dateToEpochSeconds(retryAfter)),
+          date: now.toUTCString()
+        }
+      });
+      const err = 'http error';
+      requestExecutor.fetch = jest.fn().mockImplementation(() => Promise.reject(err));
+      let errored = false;
+      await requestExecutor.retryRequest(mockRequest, mockResponse).catch(e => {
+        errored = true;
+        expect(e).toBe(err);
+      });
+      expect(errored).toBe(true);
+    });
+
+    it('should emit backoff and resume events', async () => {
+      const now = new Date();
+      const retryAfter = new Date(now.getTime() + 1000); // one second in the future
+      const requestExecutor = new DefaultRequestExecutor();
+      const mockRequest = { method: 'GET' };
+      let backoffCalled = false;
+      let resumeCalled = false;
+      const mockResponse = buildMockResponse({
+        headers: {
+          'x-rate-limit-reset' : String(dateToEpochSeconds(retryAfter)),
+          date: now.toUTCString(),
+          'x-okta-request-id': 'foo'
+        }
+      });
+      requestExecutor.fetch = jest.fn(() => Promise.resolve(mockResponse));
+      requestExecutor.on('backoff', (request, response, requestId, delayMs) => {
```

Comment:
Rather than set state here for whether these were called, a more common pattern would be to subscribe with some new spies, and then after the await below making these assertions against the spies (i.e. called once, was called with these arguments, etc).